### PR TITLE
Update error check in validateRevalidate

### PR DESCRIPTION
**Pull request by ijjk**
Applies https://github.com/vercel/next.js/pull/59822#discussion_r1433267299

Closes NEXT-1916

### CODE REVIEW

**Review by ztanner — APPROVED** [review]

File: packages/next/src/server/lib/patch-fetch.ts [hunk 1]
```diff
@@ -38,11 +38,7 @@ export function validateRevalidate(
     return normalizedRevalidate
   } catch (err: any) {
     // handle client component error from attempting to check revalidate value
-    if (
-      err &&
-      typeof err === 'object' &&
-      err.message?.includes('Invalid revalidate')
-    ) {
+    if (err instanceof Error && err.message?.includes('Invalid revalidate')) {
```
Comment (ztanner) [inline packages/next/src/server/lib/patch-fetch.ts:41]:
```suggestion
    if (err instanceof Error && err.message.includes('Invalid revalidate')) {
```

**Comment by ijjk** [comment]
## Failing test suites 
Commit: 91862e227f0c062df15b61f277e751d7b31edf17

`pnpm test test/integration/prerender-export/test/index.test.js` 

<details>
<summary>Expand output</summary>



  ● Test suite failed to run

    thrown: "Exceeded timeout of 60000 ms for a hook.
    Add a timeout value to this test to increase the timeout, if this is a long-running test. See https://jestjs.io/docs/api#testname-fn-timeout."

      217 |       })
      218 |
    > 219 |       afterAll(async () => {
          |       ^
      220 |         if (app) {
      221 |           await stopApp(app)
      222 |         }

      at afterAll (integration/prerender-export/test/index.test.js:219:7)
      at describe (integration/prerender-export/test/index.test.js:209:5)
      at integration/prerender-export/test/index.test.js:208:52
      at Object.describe (integration/prerender-export/test/index.test.js:207:1)

</details>


Read more about building and testing Next.js in [contributing.md](https://github.com/vercel/next.js/blob/canary/contributing.md).

**Comment by ijjk** [comment]
## Failing test suites 
Commit: 91862e227f0c062df15b61f277e751d7b31edf17

`pnpm test test/integration/next-image-new/app-dir/test/index.test.ts` 
- Image Component Default Tests > dev mode > should apply style inheritance for img elements but not wrapper elements

<details>
<summary>Expand output</summary>

  ● Image Component Default Tests › dev mode › should apply style inheritance for img elements but not wrapper elements

    expect(received).toBe(expected) // Object.is equality

    Expected: true
    Received: null

      1196 |         }()
      1197 |       `)
    > 1198 |       expect(imagesWithIds).toBe(true)
           |                             ^
      1199 |
      1200 |       const allSpans = await browser.eval(`
      1201 |         function foo() {

      at Object.toBe (integration/next-image-new/app-dir/test/index.test.ts:1198:29)

</details>


Read more about building and testing Next.js in [contributing.md](https://github.com/vercel/next.js/blob/canary/contributing.md).

**Review by huozhi — APPROVED** [review]

**Comment by ijjk** [comment]
# Stats from current PR

<details open>
<summary><strong>Default Build</strong></summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js ijjk/update-err-check | Change |
| - | - | - | - |
| buildDuration | 13.3s | 13.2s | N/A |
| buildDurationCached | 7.5s | 6.3s | N/A |
| nodeModulesSize | 200 MB | 200 MB | N/A |
| nextStartRea..uration (ms) | 431ms | 430ms | N/A |

</details>

<details>
<summary><strong>Client Bundles (main, webpack)</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js ijjk/update-err-check | Change |
| - | - | - | - |
| 193.HASH.js gzip | 181 B | 182 B | N/A |
| 3f784ff6-HASH.js gzip | 53.3 kB | 53.3 kB | ✓ |
| 433-HASH.js gzip | 28.2 kB | 28.3 kB | N/A |
| framework-HASH.js gzip | 45.2 kB | 45.2 kB | ✓ |
| main-app-HASH.js gzip | 239 B | 242 B | N/A |
| main-HASH.js gzip | 31.7 kB | 31.7 kB | N/A |
| webpack-HASH.js gzip | 1.7 kB | 1.7 kB | N/A |
| Overall change | 98.5 kB | 98.5 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js ijjk/update-err-check | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js ijjk/update-err-check | Change |
| - | - | - | - |
| _app-HASH.js gzip | 194 B | 195 B | N/A |
| _error-HASH.js gzip | 183 B | 181 B | N/A |
| amp-HASH.js gzip | 504 B | 502 B | N/A |
| css-HASH.js gzip | 321 B | 321 B | ✓ |
| dynamic-HASH.js gzip | 2.5 kB | 2.5 kB | N/A |
| edge-ssr-HASH.js gzip | 255 B | 253 B | N/A |
| head-HASH.js gzip | 350 B | 349 B | N/A |
| hooks-HASH.js gzip | 369 B | 369 B | ✓ |
| image-HASH.js gzip | 4.28 kB | 4.28 kB | N/A |
| index-HASH.js gzip | 255 B | 256 B | N/A |
| link-HASH.js gzip | 2.61 kB | 2.61 kB | ✓ |
| routerDirect..HASH.js gzip | 312 B | 311 B | N/A |
| script-HASH.js gzip | 385 B | 383 B | N/A |
| withRouter-HASH.js gzip | 307 B | 308 B | N/A |
| 1afbb74e6ecf..834.css gzip | 106 B | 106 B | ✓ |
| Overall change | 3.4 kB | 3.4 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js ijjk/update-err-check | Change |
| - | - | - | - |
| _buildManifest.js gzip | 483 B | 484 B | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js ijjk/update-err-check | Change |
| - | - | - | - |
| index.html gzip | 528 B | 527 B | N/A |
| link.html gzip | 540 B | 539 B | N/A |
| withRouter.html gzip | 523 B | 523 B | ✓ |
| Overall change | 523 B | 523 B | ✓ |

</details>

<details>
<summary><strong>Edge SSR bundle Size</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js ijjk/update-err-check | Change |
| - | - | - | - |
| edge-ssr.js gzip | 93.7 kB | 93.7 kB | N/A |
| page.js gzip | 147 kB | 147 kB | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Middleware size</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js ijjk/update-err-check | Change |
| - | - | - | - |
| middleware-b..fest.js gzip | 624 B | 624 B | ✓ |
| middleware-r..fest.js gzip | 151 B | 151 B | ✓ |
| middleware.js gzip | 37.4 kB | 37.4 kB | N/A |
| edge-runtime..pack.js gzip | 1.92 kB | 1.92 kB | ✓ |
| Overall change | 2.7 kB | 2.7 kB | ✓ |

</details>

<details>
<summary><strong>Next Runtimes</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js ijjk/update-err-check | Change |
| - | - | - | - |
| app-page-exp...dev.js gzip | 168 kB | 168 kB | N/A |
| app-page-exp..prod.js gzip | 94.2 kB | 94.2 kB | N/A |
| app-page-tur..prod.js gzip | 94.9 kB | 94.9 kB | N/A |
| app-page-tur..prod.js gzip | 89.4 kB | 89.4 kB | N/A |
| app-page.run...dev.js gzip | 138 kB | 138 kB | N/A |
| app-page.run..prod.js gzip | 88.7 kB | 88.7 kB | N/A |
| app-route-ex...dev.js gzip | 24.1 kB | 24.1 kB | N/A |
| app-route-ex..prod.js gzip | 16.7 kB | 16.7 kB | ✓ |
| app-route-tu..prod.js gzip | 16.7 kB | 16.7 kB | ✓ |
| app-route-tu..prod.js gzip | 16.3 kB | 16.3 kB | N/A |
| app-route.ru...dev.js gzip | 23.5 kB | 23.5 kB | N/A |
| app-route.ru..prod.js gzip | 16.3 kB | 16.3 kB | N/A |
| pages-api-tu..prod.js gzip | 9.37 kB | 9.37 kB | ✓ |
| pages-api.ru...dev.js gzip | 9.64 kB | 9.64 kB | ✓ |
| pages-api.ru..prod.js gzip | 9.37 kB | 9.37 kB | ✓ |
| pages-turbo...prod.js gzip | 21.9 kB | 21.9 kB | ✓ |
| pages.runtim...dev.js gzip | 22.5 kB | 22.5 kB | ✓ |
| pages.runtim..prod.js gzip | 21.9 kB | 21.9 kB | ✓ |
| server.runti..prod.js gzip | 49.4 kB | 49.4 kB | ✓ |
| Overall change | 178 kB | 178 kB | ✓ |

</details>

<details>
<summary><strong>Diff details</strong></summary>

<details>
<summary>Diff for <strong>page.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-page-exp..ntime.dev.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-page-exp..time.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-page-tur..time.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-page-tur..time.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-page.runtime.dev.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-page.runtime.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-route-ex..ntime.dev.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-route-ex..time.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-route-tu..time.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-route-tu..time.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-route.runtime.dev.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-route.ru..time.prod.js</strong></summary>

Diff too large to display
</details>

</details>

</details>
Commit: 6031dd93046ad66ad4676ce28e421283ac0c6932

**Pull request merged**